src/freertos_drivers/tinyusb/TinyUsbCdc.hxx

```diff
@@ -0,0 +1,99 @@
+/** \copyright
+ * Copyright (c) 2023, Balazs Racz
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are  permitted provided that the following conditions are met:
+ * 
+ *  - Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  - Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * \file TinyUsbCdc.hxx
+ * Base class for implementing a CDC Device driver using the TinyUsb stack.
+ *
+ * @author Balazs Racz
+ * @date 13 Nov 2023
+ */
+
+#ifndef _FREERTOS_DRIVERS_TINYUSB_TINYUSBCDC_HXX_
+#define _FREERTOS_DRIVERS_TINYUSB_TINYUSBCDC_HXX_
+
+#include "Devtab.hxx"
+#include "os/OS.hxx"
+#include "utils/Singleton.hxx"
+
+class TinyUsbCdc : public Node, public Singleton<TinyUsbCdc> {
```

Comment:
It's not super easy to create multiple CDC endpoints. For reference, the TivaUsbCdc driver doesn't support it either. These are the steps that need to be taken before we can instantiate a second object:
- change the tinyusbconfig.h (which will have a memory footprint effect)
- change the USB descriptors
- update how the thread is started / running
- update all callbacks to be routed to the right instance
- figure out what happens on the host OS if there is no second object but the descriptor says there should be

Ultimately I think having two CDCs is a feature request that's not high priority, and at this point it's more important to prevent accidentally creating two instances, which the singleton will do fine.